app/controllers/posts_controller.rb

```diff
@@ -18,10 +18,29 @@ def create
     if @post.save
       redirect_to root_path
     else
-      render 'posts/new'
+      render :new
     end
   end
 
+  def edit
+    unless current_user.id == @post.user_id
```

Comment:
unless文は、実行するコードが1行なら後置で記述してあげたほうが読みやすいです。
```suggestion
    redirect_to root_path unless current_user.id == @post.user_id
```